ui/admin/app/routes/scopes/scope/roles/role.js

```diff
@@ -40,4 +41,27 @@ export default class ScopesScopeRolesRoleRoute extends Route {
       this.router.replaceWith(transition.to.name, role.scopeID, role.id);
     }
   }
+
+  // =actions
+
+  /**
+   * Event to determine whether the loading template should be shown.
+   * Only show the loading template during initial loads for specified
+   * routes. Don't show it when a user is just searching or
+   * filtering on the same page as it can be jarring.
+   * @param transition
+   * @returns {boolean}
+   */
+  @action
+  loading(transition) {
+    const to = transition.to?.name;
+    const from = transition.from?.name;
+    return (
+      (to === 'scopes.scope.roles.role.index' ||
+        to === 'scopes.scope.roles.role.manage-scopes.manage-custom-scopes' ||
+        to === 'scopes.scope.roles.role.manage-scopes.manage-org-projects' ||
+        to === 'scopes.scope.roles.role.scopes') &&
+      from !== to
```

Comment:
Not blocking but should some of these have been in another `loading` action in the manage scopes route instead?

Reply:
i attempted to handle the loading in the separate routes however, the issue is that when transitioning in between routes it still uses the loading action in the `role` route which was complicating things so decided to handle it all together.